packages/generator/src/name-formatting-util.ts

```diff
@@ -0,0 +1,138 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import {
+  reservedObjectPrototypeKeywords,
+  reservedVdmKeywords
+} from './name-formatting-reserved-key-words';
+
+type Separator = '-' | '_';
+const MAXIMUM_NUMBER_OF_SUFFIX = 1000;
+
+export interface FindUniqueName {
```

Comment:
The thing with just an array of names is that his becomes tricky for the integer suffixes. For example you start with [Name,NameType,NameTypeForceMandatory] but name is occupied. I thought it would be nicer to have [Name_1,Name_1Type,Name_1TypeForceMandatory] instead of [Name_1,NameType_1,NameTypeForceMandatory_1]. But to put the `_1` in the right place you need to know how to construct the related names from the initial input.

Reply:
Hmm ok, I see. Could we then just use the fact that these are always suffixes (I mean how probable is it that there will be other ways of constructing names, that are somehow related...)? What do you think about that?

Reply:
I changed it to a suffix handling case.